src/main/java/model/structures/GraphImpl.java

```diff
@@ -0,0 +1,144 @@
+    }
+
+    public int getIterationStep()
+    {
+        return iterationStep;
+    }
+
+    public GraphImpl(ArrayList<Node> nodeArrayList, int[][] adjacencyMatrix)
+    {
+        this.adjacencyMatrix = adjacencyMatrix;
+        graphSize = nodeArrayList.size();
+
+        nodes = new HashMap<>();
+        edgeWageSum = 0;
+
+        for(Node node : nodeArrayList)
+            nodes.put(node.getId(), node);
+
+        for (int[] matrix : adjacencyMatrix)
+            for (int j = 0; j < adjacencyMatrix.length; j++)
+                edgeWageSum += matrix[j];
+
+        edgeWageSum /= 2;
+
+    }
+
+    private void generatePeople(long seedForHabitats, int minimalNumberOfHabitats, int maximumNumberOfHabitats, int graphSize, int infectingParameter)
+    {
+        Random random = new Random(seedForHabitats);
+        nodes = new HashMap<>();
+
+        for(int i = 0; i < graphSize; i++)
+        {
+            //generowanie nodow
+            nodes.put(i, new Node(i, Math.min(maximumNumberOfHabitats, random.nextInt((maximumNumberOfHabitats - minimalNumberOfHabitats)) + minimalNumberOfHabitats), infectingParameter));
+        }
+    }
+
+    public void nextStep(int iterationNumber)
+    {
+        //aktualizacja zarazonych w nodach
+        for(Node n : nodes.values())
+            n.updateIllnessCases(iterationNumber);
+
+        //zmiana wartosci adjacencyMatrix
+        adjacencyMatrix = ModelGenerator.generateAdjacencyMatrix(graphSize, edgeWageSum, randomForAdjacencyMatrix);
+    }
+
+    public int getWage(int firstNodeId, int secondNodeId)
+    {
+        return adjacencyMatrix[firstNodeId][secondNodeId];
+    }
+
+    public Graph getSubgraphWithUnvisitedNodes()
+    {
+        HashMap<Integer, Node> subgraphNodes = new HashMap<>();
+
+        for(Node n : nodes.values())
+        {
+            if(!n.getIsVaccinated())
+            {
+                subgraphNodes.put(n.getId(), n);
+            }
+        }
+
+        return new GraphImpl(adjacencyMatrix, subgraphNodes);
+    }
+
+    public Node getNode(int id)
+    {
+        return nodes.get(id);
+    }
+
+    public HashMap<Integer, Node> getNodes()
+    {
+        return nodes;
+    }
+
+    public int[][] getAdjacencyMatrix() {
+        return adjacencyMatrix;
+    }
+
+    @Override
+    public GraphPath getRandomPath() {
+        ArrayList<Node> pathNodes = new ArrayList<>();
+
+        for(Node n : nodes.values())
+            if(!n.getIsVaccinated())
+                pathNodes.add(new Node(n));
+
+        Collections.shuffle(pathNodes);
+
+        GraphPathImpl graphPath = new GraphPathImpl(adjacencyMatrix);
+
+        for(int i = 0; i < pathNodes.size(); i++)
+            graphPath.addToPath(pathNodes.get(i), i != 0 ? getWage(pathNodes.get(i).getId(), graphPath.getLastNode().getId()) : 0);
+
+        return graphPath;
+    }
+
+    @Override
+    public Graph getUpdatedGraphWithoutNode(Node nodeToBeRemoved) {
+        nodeToBeRemoved.visit(iterationStep);
```

Comment:
Po co ten node ma być visit? Co to zmienia jeśli i tak jest usuwany?

Reply:
tak, jak rozmawialiśmy metoda visit() oznacza, że odwiedzamy dany nodę, aby dokonać dostawy szczepionki

Reply:
Poza tym on nie jest stricte usuwany, nie będzie go w instancji podgrafu